addon/connection.cc

```diff
@@ -87,30 +93,18 @@ Napi::Value Connection::Execute(const Napi::CallbackInfo &info) {
                                         static_cast<int>(ResultFormat::ARRAY)));
       }
     }
-
-    AsyncExecutor *wk = new AsyncExecutor(env, query, connection, deferred,
-                                          forceMaterializedValue,
-                                          rowResultFormatValue, results);
-    wk->Queue();
+    async_executors->push_back(duckdb::make_unique<NodeDuckDB::AsyncExecutor>(
+        info, query, connection, forceMaterializedValue, rowResultFormatValue));
+    async_executors->back()->Execute();
```

Comment:
Do we ever clean these up?

Reply:
connection's destructor should destroy these